pinot-core/src/test/java/com/linkedin/pinot/core/startree/TestStarTreeIntegrationTest.java

```diff
@@ -93,41 +99,70 @@ void tearDown()
         "select sum(m1) from T",
-        "select sum(m1) from T where d1 = 'd1-v2' group by d2"
+        "select sum(m1) from T where d1 = 'd1-v1'",
+        "select sum(m1) from T where d1 <> 'd1-v1'",
+        "select sum(m1) from T where d1 between 'd1-v1' and 'd1-v3'",
+        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2')",
+        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') and d2 not in ('d2-v1')",
+        "select sum(m1) from T group by d1",
+        "select sum(m1) from T group by d1, d2",
+        "select sum(m1) from T where d1 = 'd1-v2' group by d1",
+        "select sum(m1) from T where d1 between 'd1-v1' and 'd1-v3' group by d2",
+        "select sum(m1) from T where d1 = 'd1-v2' group by d2, d3",
+        "select sum(m1) from T where d1 <> 'd1-v1' group by d2",
+        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') group by d2",
+        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') and d2 not in ('d2-v1') group by d3",
+        "select sum(m1) from T where d1 in ('d1-v1', 'd1-v2') and d2 not in ('d2-v1') group by d3, d4"
     };
 
     // Test against all metric columns, instead of just the aggregation column in the query.
     List<String> metricNames = _schema.getMetricNames();
+    SegmentMetadata segmentMetadata = _segment.getSegmentMetadata();
 
     for (int i = 0; i < queries.length; i++) {
       Pql2Compiler compiler = new Pql2Compiler();
       BrokerRequest brokerRequest = compiler.compileToBrokerRequest(queries[i]);
 
-      double[] expectedSums = computeSumUsingRawDocs(metricNames, brokerRequest);
-      double[] actualSums = computeSumUsingAggregatedDocs(metricNames, brokerRequest);
+      FilterQueryTree filterQueryTree = RequestUtils.generateFilterQueryTree(brokerRequest);
+      Assert.assertTrue(RequestUtils.isFitForStarTreeIndex(segmentMetadata, filterQueryTree, brokerRequest));
 
-      Assert.assertEquals(expectedSums.length, actualSums.length, "Mis-match in number of metrics");
-      for (int j = 0; j < expectedSums.length; j++) {
-        Assert.assertEquals(expectedSums[j], actualSums[j],
-            "Mis-match result for metric: " + metricNames.get(j) + " Acutal: " + actualSums[j] + " Expected: "
-                + expectedSums[j]);
+      Map<String, double[]> expectedResult = computeSumUsingRawDocs(metricNames, brokerRequest);
+      Map<String, double[]> actualResult = computeSumUsingAggregatedDocs(metricNames, brokerRequest);
+
+      Assert.assertEquals(expectedResult.size(), actualResult.size(), "Mis-match in number of groups");
+      for (Map.Entry<String, double[]> entry : expectedResult.entrySet()) {
+        String expectedKey = entry.getKey();
+        Assert.assertTrue(actualResult.containsKey(expectedKey));
+
+        double[] expectedSums = entry.getValue();
+        double[] actualSums = actualResult.get(expectedKey);
+
+        for (int j = 0; j < expectedSums.length; j++) {
+          Assert.assertEquals(expectedSums[j], actualSums[j],
```

Comment:
The convention is to put actual value as first argument and expected as second. In the error message, no need to explicitly log actual and expected value since they will already show up.